src/prototype_room_market.js

```diff
@@ -169,15 +175,41 @@ Room.prototype.sendPowerOwnRooms = function() {
   }
 };
 
+Room.prototype.sendEnergyToMyRooms = function() {
+  if ((_.size(Memory.needEnergyRooms) > 0) && (_.size(Memory.canHelpRooms) > 0) &&
+    _.includes(Memory.canHelpRooms, this.name) && !_.includes(Memory.needEnergyRooms, this.name) &&
+    (this.terminal.store[RESOURCE_ENERGY] > config.terminal.minEnergyAmount)) {
+    const diffNeedEnergyRooms = _.difference(Memory.needEnergyRooms, Memory.canHelpRooms);
+    const needEnergyRooms = _.size(diffNeedEnergyRooms) > 0 ? diffNeedEnergyRooms : Memory.needEnergyRooms;
```

Comment:
To be honest I wouldn't to that. We should rely on, that `Memory.needEnergyRooms` and the ones which actually need energy. (Add they shouldn't be in the other list)

Reply:
you mean `!_.includes(Memory.needEnergyRooms, this.name) &&`
as i was working on the improvements, it did happen some times... i will remove